### PR TITLE
os/bluestore: make verify_csum catch unsupported csum type error

### PR DESCRIPTION
**Pull request by xiexingguo**
And update related unit-testing cases.

### CODE REVIEW

**Comment by xiexingguo** [comment]
@liewegas Upgrade https://github.com/ceph/ceph/pull/9498/commits/ff27e14c160397a1f3bdd06255b3d75ebf10726a here.

File: src/os/bluestore/bluestore_types.cc [hunk 1]
```diff
@@ -668,23 +668,38 @@ void bluestore_blob_t::calc_csum(uint64_t b_off, const bufferlist& bl)
   }
 }
 
-int bluestore_blob_t::verify_csum(uint64_t b_off, const bufferlist& bl) const
+int bluestore_blob_t::verify_csum(uint64_t b_off, const bufferlist& bl,
+  int &b_bad_off) const
```
Comment (liewegas) [inline src/os/bluestore/bluestore_types.cc]: style nit: can you make this int* instead of int& to indicate it is an output argument?

**Comment by liewegas** [comment]
Also, please combine the unittest update in the same commit as the change.

Thanks!

**Comment by liewegas** [comment]
Otherwise it looks good!

**Comment by xiexingguo** [comment]
Updated as you suggested.

**Pull request merged**